app/helpers/common/priceFormatter.js

```diff
@@ -9,10 +9,13 @@ export const formatNumber = ({
 
 export const formatPrice = ({ value, minimumFractionDigits = 2, maximumFractionDigits = 4 }) => {
   if (minimumFractionDigits === 0) {
-    if (Number.isInteger(Number.parseFloat(value))) {
-      return formatNumber({ value, maximumFractionDigits });
+    if (!Number.isNaN(Number(value)) && !Number.isNaN(parseFloat(value))) {
```

Comment:
it might work, but the logic here is not easy to follow, would benefit from a refactor

Reply:
done few changes as a part of refactoring